c10/core/Device.h

```diff
@@ -209,7 +222,7 @@ struct hash<c10::Device> {
     // sake.
     uint32_t bits = static_cast<uint32_t>(static_cast<uint8_t>(d.type()))
             << 16 |
-        static_cast<uint32_t>(static_cast<uint8_t>(d.index()));
+        static_cast<uint32_t>(static_cast<uint16_t>(d.index()));
```

Comment:
Use static_cast<uint32_t>(d.index()) directly?

Reply:
This would cause problems when the index is -1 (see the code comment above).